design/list_command_design.md

```diff
@@ -0,0 +1,289 @@
+# Dockerless CLI - `list` action
+
+This document is the design document for the list CLI command for the Dockerless project. The main functionality for the list CLI command has been implemented. Now all that is left is to implement the following flags:
+
+1. `--[no]inspect`
+2. `-k, --kind`
+3. `-l, --limit`
+4. `-o, --offset`
+5. `--role`
+6. `-s, --search`
+
+In additional to implementing the above flags, we will need to also address the following:
+
+1. Ensure UTs and integration tests exist and are GA quality
+2. Improve error handling experience
+3. Write proper documentation in `README.md` of the repository
+
+```bash
+Name
+----
+  list - Lists all available resources beloging to this account
+
+
+Usage
+----
+  conjur [global options] list [arguments...]
+
+
+Global options
+----
+  -h, --help         Show this help message and exit
+  -d, --debug				 Enable debugging output
+
+       
+Commands
+----
+--[no-]inspect       Show full object information
+
+-k, --kind=arg       Filter by kind (default: none)
+
+-l, --limit=arg      Maximum number of records to return (default: none)
+
+-o, --offset=arg     Offset to start from (default: none)
+
+--role=ROLE          Role to act as. By default, the current logged-in role is used. (default: none)
```

Comment:
why we dont define here shortcut as well ? 
and its need to be declared better how the role suppose to look like (you need to add the org before the role)

Reply:
@eladkug you are right
shortcut should be -r

Reply:
@InbalZilberman -r may be a problem since its deprecated for python (and still exist in ruby cli)
`-r, --[no-]raw-annotations - Show annotations in 'raw' format`

Reply:
lets go with -r for role and when we add raw-annotations i think we will add it as -a

Reply:
@InbalZilberman why do we need the `=` between each key and value. From what I have seen it is usually `-<flag> <flag-value>`